bpf/lib/tunnel.h

```diff
@@ -83,4 +88,135 @@ tunnel_vni_to_sec_identity(__be32 vni)
 	return bpf_ntohl(vni) >> 8;
 }
 
+/*
+ * Returns true if the skb associated with data pointers is a vxlan encapsulated
+ * packet.
+ *
+ * The determination is made by comparing the UDP destination port with
+ * the tunnel_port provided to the function.
+ */
+static __always_inline bool
+tunnel_skb_is_vxlan_v4(const void *data, const void *data_end,
+		       const struct iphdr *ipv4, const __u16 tunnel_port)
+{
+	struct udphdr *udp = NULL;
+	__u32 l3_size = 0;
+
+	if (ipv4->protocol != IPPROTO_UDP)
+		return false;
+
+	l3_size = ipv4->ihl * 4;
```

Comment:
```suggestion
	l3_size = ipv4_hdrlen(ipv4);
```

(please check for other occurrences)

Reply:
Going to resolve this for now. I have a personal endeavor to make this situation better, once I unravel this stuff in a separate PR I will replace suggested line above (the suggestion is good).